### ISSUE TITLE
Problems with all installation methods (or improve README)

### ISSUE THREAD

**Issue by vits** [humbug/php-scoper#369]
## Bug report

| Question    | Answer
| ------------| ---------------
| Box version | 0.13.0
| PHP version | PHP 7.3.13 (cli) (built: Dec 18 2019 16:08:06) ( NTS )
| Platform with version | MacOS Catalina + brew
| Github Repo | -

I found it very hard to get php-scoper installed and running. I have problems with all listed installation methods except Phive, which I didn't try.

PHAR version has problem with scoper.inc.php as described in #368.

`composer global require humbug/php-scoper` has problem with `laravel\valet` installed because of `symfony\console` conflict

```
 Problem 1
    - Installation request for humbug/php-scoper ^0.13.0 -> satisfiable by humbug/php-scoper[0.13.0].
    - Conclusion: remove symfony/console v5.0.2
    - Conclusion: don't install symfony/console v5.0.2
    - humbug/php-scoper 0.13.0 requires symfony/console ^3.2 || ^4.0 -> satisfiable by symfony/console[3.2.x-dev, 3.3.x-dev, 3.4.x-dev, 4.0.x-dev, 4.1.x-dev, 4.2.x-dev, 4.3.x-dev, 4.4.x-dev, v3.2.0, v3.2.0-BETA1, v3.2.0-RC1, v3.2.0-RC2, v3.2.1, v3.2.10, v3.2.11, v3.2.12, v3.2.13, v3.2.14, v3.2.2, v3.2.3, v3.2.4, v3.2.5, v3.2.6, v3.2.7, v3.2.8, v3.2.9, v3.3.0, v3.3.0-BETA1, v3.3.0-RC1, v3.3.1, v3.3.10, v3.3.11, v3.3.12, v3.3.13, v3.3.14, v3.3.15, v3.3.16, v3.3.17, v3.3.18, v3.3.2, v3.3.3, v3.3.4, v3.3.5, v3.3.6, v3.3.7, v3.3.8, v3.3.9, v3.4.0, v3.4.0-BETA1, v3.4.0-BETA2, v3.4.0-BETA3, v3.4.0-BETA4, v3.4.0-RC1, v3.4.0-RC2, v3.4.1, v3.4.10, v3.4.11, v3.4.12, v3.4.13, v3.4.14, v3.4.15, v3.4.16, v3.4.17, v3.4.18, v3.4.19, v3.4.2, v3.4.20, v3.4.21, v3.4.22, v3.4.23, v3.4.24, v3.4.25, v3.4.26, v3.4.27, v3.4.28, v3.4.29, v3.4.3, v3.4.30, v3.4.31, v3.4.32, v3.4.33, v3.4.34, v3.4.35, v3.4.36, v3.4.4, v3.4.5, v3.4.6, v3.4.7, v3.4.8, v3.4.9, v4.0.0, v4.0.0-BETA1, v4.0.0-BETA2, v4.0.0-BETA3, v4.0.0-BETA4, v4.0.0-RC1, v4.0.0-RC2, v4.0.1, v4.0.10, v4.0.11, v4.0.12, v4.0.13, v4.0.14, v4.0.15, v4.0.2, v4.0.3, v4.0.4, v4.0.5, v4.0.6, v4.0.7, v4.0.8, v4.0.9, v4.1.0, v4.1.0-BETA1, v4.1.0-BETA2, v4.1.0-BETA3, v4.1.1, v4.1.10, v4.1.11, v4.1.12, v4.1.2, v4.1.3, v4.1.4, v4.1.5, v4.1.6, v4.1.7, v4.1.8, v4.1.9, v4.2.0, v4.2.0-BETA1, v4.2.0-BETA2, v4.2.0-RC1, v4.2.1, v4.2.10, v4.2.11, v4.2.12, v4.2.2, v4.2.3, v4.2.4, v4.2.5, v4.2.6, v4.2.7, v4.2.8, v4.2.9, v4.3.0, v4.3.0-BETA1, v4.3.0-BETA2, v4.3.0-RC1, v4.3.1, v4.3.2, v4.3.3, v4.3.4, v4.3.5, v4.3.6, v4.3.7, v4.3.8, v4.3.9, v4.4.0, v4.4.0-BETA1, v4.4.0-BETA2, v4.4.0-RC1, v4.4.1, v4.4.2].
```

However it is possible to install it if `laravel\valet` is removed, and then valet installs fine after php-scoper, so problem solved for now and it would be fine from beginning if next method with `composer-bin-plugin` worked. Unfortunately it doesn't:

```
Problem 1
    - Installation request for humbug/php-scoper ^0.13.0 -> satisfiable by humbug/php-scoper[0.13.0].
    - humbug/php-scoper 0.13.0 requires jetbrains/phpstorm-stubs dev-master -> satisfiable by jetbrains/phpstorm-stubs[dev-master] but these conflict with your requirements or minimum-stability.
```

`"minimum-stability": "dev"` is set both in global `composer.json` and in project, so I suppose this is somehow related to `composer-bin-plugin` ... and just now searching for some solution or confirmation of this I found this bamarni/composer-bin-plugin#49. It could have helped much, if those two additional command lines were added to php-scoper README file.

Sorry, hope I don't sound too negative, just a bit frustrated with all these unexpected hurdles.

**Comment by theofidry** [comment]
#368 is a real issue and I really need to take a look at it.

>composer global require humbug/php-scoper has problem with laravel\valet installed because of symfony\console conflict

It should not be too hard to allow `symfony/console 5.x`, PRs are welcome :)

>https://github.com/bamarni/composer-bin-plugin/issues/49

Indeed: but as per the issue, PRs are welcomed

>Sorry, hope I don't sound too negative, just a bit frustrated with all these unexpected hurdles.

No problem but keep in mind 2/3 of your issues can easily be fixed with a PR without having any deep understanding of the projects, so it's more a contribution issue than anything :)

**Comment by vits** [comment]
Sure, #368 is real issue here and everything else is just my misadventures :). However I kind of disagree with you that there must be PR or else you just ignore simple documentation improvements as described in composer-bin-plugin issue. I guess it would be less work for you to just add those two lines in README than to write comments about PR. :) Contribution is good and I kind of feel contributing right now by exploring and describing these issues, but at the end this is your project and it just needs simple (and known to you) improvement to documentation. Are you really ok to let other potential users of php-scoper (which is great tool btw! thank you!) get into same problems as me just because nobody is willing to submit PR?

`symfony\console` issue is another case, I understand that it needs some work and testing, so PR is the right way.

**Comment by theofidry** [comment]
That's true, but at the end of the day it's a question of what you want to focus on when you are working on OSS. Whilst questionable, first and foremost, I solve _my_ issues first. Then I do put some efforts in trying to be responsive to issues & PRs. But for the rest it's mostly on what I'm currently feeling motivating on, because the above is already quite taxing in time & energy.

**Comment by theofidry** [comment]
Closed via #373 and #372

**Comment by arku31** [comment]
I still had this issue.
I have resolved it by having
```"minimum-stability": "dev",```
 in the global composer folder -  ~/.composer/composer.json
